http_service/tests/test_utils_integration.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import sys
+import time
+from datetime import timedelta
+
+from bugbug_http.utils import ReadthroughTTLCache
+
+# to load bugbug_http.utils, when running from project root
+sys.path.append(".")
+
+
+def integration_test_cache_thread():
```

Comment:
This doesn't have to be an integration test, it could be a normal test (so, you can move it to test_utils.py).

Reply:
My thinking at the time was that its better to have this be separate script rather unit test. How does the below justification sound to you? I'm happy to re-evaluate if you'd like me to

The advantage of having this be a seperate script rather than a unit test is that this takes real time to run - as written, it'd make every single unit test run take an extra seven seconds, which means noticeably longer test runs for every test run for this project! But, as a separate script, we can exclude it from test runs and not have that problem.

Since the cache expects a value on the order of seconds for its TTL, I'm not sure if there is a way to have this test run in a negligible amount of time, which is what I expect would be ideal for a unit test.

Reply:
Wait nevermind - I can write this as a unit test by mocking out sleep. Thanks!

Reply:
On second thought, it might be possible to write this as a unit test with proper mocking - it'd be a bit tricky though

Reply:
You could mock `time.sleep`, dividing the sleep time by 10.